www/js/GenWidget.js

```diff
@@ -0,0 +1,158 @@
+/* global Widget, NNE, HTMLElement */
+/*
+  -----------
+     info
+  -----------
+
+  This Code Generator Widget extends the base Widget class to add methods
+  for more easily creating CSS generator Widgets.
+
+  -----------
+     usage
+  -----------
+
+  // this class inherits all the properties/methods of the base Widget class
+  // refer to www/js/Widget.js to see what those are
+  // or take a look at the wiki...
+  // https://github.com/netizenorg/netnet.studio/wiki/Creating-Widgets
+
+  this.codeField = this.createCodeField({
+    value: 'font-size 24px;',
+    change: (e) => this._codeFieldUpdate(e)
+  })
+
+  this.slider = w.createSlider({
+    background: '#f00',
+    min: 0,
+    max: 100,
+    label: 'PX'
+    change: (e) => this._fontSliderUpdate(e)
+  })
+
+*/
+class CodeField extends HTMLElement {
+  connectedCallback (opts) {
+    this.innerHTML = `<div>
+        <button>insert</button>
+        <input type="text">
+    </div>`
+
+    this.input = this.querySelector('input')
+    this.insertButton = this.querySelector('button')
```

Comment:
these two lines aren't necessary, u're not using `this.input` or `this.inserButton` anywhere, we can remove this.